pkg/apis/core/validation/validation.go

```diff
@@ -4064,8 +4065,16 @@ func ValidateService(service *core.Service) field.ErrorList {
 		}
 	}
 
-	allErrs = append(allErrs, validateServiceExternalTrafficFieldsValue(service)...)
+	//if an ipfamily provided then it has to be one of the supported values
+	// note:
+	// - we don't validate service.Spec.IPFamily is supported by the cluster
+	// - we don't validate service.Spec.Cluster is within a range supported by the cluster
+	// both of these validations are done by the ipallocator
+	if service.Spec.IPFamily != nil && !supportedServiceIPFamily.Has(string(*service.Spec.IPFamily)) {
```

Comment:
we have been very specific on doing gate specific checks on validation. We have been explicitly saying no, we should not do that. Also the field itself is not required even when the gate is removed, the cluster will assign default. Did i miss understand something?

Reply:
Jordan agreed that the valid use of gates in validation was this pattern (in the side thread :)

Since validation runs AFTER defaulting, the only way `nil` would be seen in validation is either a) the gate is off or b) a bug.  This is what I mean about required vs optional.  It is actually required on read, optional on write.  BTW, I have a consult with apimachinery folks in 30 minutes :)

Long-term leaving a nil-check bypass here means we won't find that bug (as easily).